astropy/coordinates/representation.py

```diff
@@ -221,6 +227,136 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def __eq__(self, other):
+        """Check for equality with another representation.
+
+        If either ``self`` or ``other`` are not scalar, this returns a boolean
+        array, just like is the case for ``ndarray``.  Elements are `True` only
+        if all components of that element are exactly equal.
+        """
+        try:
+            # this ensures that other can be compared sensibly to self.
+            # note that it immediately returns if the classes are the same.
+            other = other.represent_as(self.__class__)
+        except Exception:
+            return False
+
+        # Ensure we return a bool or array of bool as appropriate.
+        return functools.reduce(np.logical_and,
+                                (getattr(self, component) ==
+                                 getattr(other, component)
+                                 for component in self.components))
+
+    def __ne__(self, other):
+        return np.logical_not(self.__eq__(other))
+
+    def __mul__(self, other):
+        values = (getattr(self, component) for component in self.components)
+        return self.__class__(*(
+            (value if issubclass(cls, Angle) else value * other)
+            for (component, cls), value in zip(self.attr_classes.items(),
+                                               values)))
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        values = (getattr(self, component) for component in self.components)
+        return self.__class__(*(
+            (value if issubclass(cls, Angle) else value / other)
+            for (component, cls), value in zip(self.attr_classes.items(),
+                                               values)))
+
+    def __div__(self, other):
+        return self.__truediv__(other)
+
+    def __abs__(self):
+        return np.sqrt(functools.reduce(
+            operator.add, (getattr(self, component)**2
+                           for component, cls in self.attr_classes.items()
+                           if not issubclass(cls, Angle))))
+
+
+    def _operation(self, operator, other, reverse=False):
+        result = self.to_cartesian()._operation(operator, other, reverse)
```

Comment:
I'm not sure if it's a good idea to use a variable with the same name as a module you imported. Maybe `op` instead?

However this is not really important.